drivers/adc/adc_stm32.c

```diff
@@ -607,9 +607,21 @@ static int start_read(const struct device *dev,
 	defined(CONFIG_SOC_SERIES_STM32WLX)
 	/* STM32G0 in "not fully configurable" sequencer mode */
 	LL_ADC_REG_SetSequencerChannels(adc, channel);
+	LL_ADC_REG_SetSequencerConfigurable(adc, LL_ADC_REG_SEQ_CONFIGURABLE);
```

Comment:
this is going to change for stm32G0 too, but this device is preferably configured with 
LL_ADC_REG_SetSequencerConfigurable(adc, **LL_ADC_REG_SEQ_FIXED**);